sourced/ml/cmd/args.py

```diff
@@ -17,6 +25,30 @@ def _get_help_string(self, action):
         return super()._get_help_string(action)
 
 
+def handle_input_arg(input_arg: str,
+                     filter_arg: Optional[str] = DEFAULT_FILTER_ARG,
+                     log: Optional[logging.Logger] = None):
+    """
+    Process input arguments and return an iterator over input files.
+
+    :param input_arg: Directory path to scan or `-` to get file paths from stdin
+    :param filter_arg: File name glob selector for input directory. Ignored if `input_arg` is \
+        equal to `-`.
+    :param log: Logger if you want to log handling process.
+    :return: an iterator over input files.
+    """
+    log = log.info if log else (lambda *x: None)
+    if input_arg == "-":
+        log("Reading file paths from stdin.")
+        for line in sys.stdin:
+            yield line.strip()
+    else:
+        log("Scanning %s", input_arg)
+        if filter_arg:
```

Comment:
If there is no filter arg we should `return input_arg` (provided by `nargs="+"`)